components/server/src/auth/bearer-authenticator.ts

```diff
@@ -79,19 +80,51 @@ export class BearerAuth {
         if (!token) {
             throw createBearerAuthError("missing Bearer token");
         }
-        const userAndToken = await this.userDB.findUserByGitpodToken(token, GitpodTokenType.API_AUTH_TOKEN);
-        if (!userAndToken) {
-            throw createBearerAuthError("invalid Bearer token");
-        }
 
-        // hack: load the user again to get ahold of all identities
-        // TODO(cw): instead of re-loading the user, we should properly join the identities in findUserByGitpodToken
-        const user = (await this.userDB.findUserById(userAndToken.user.id))!;
+        let user: User;
```

Comment:
This is a good sign that we might want to extract the following code into a `tryParseToken` function.